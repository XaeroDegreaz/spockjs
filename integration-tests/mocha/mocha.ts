import { sync as run } from 'cross-spawn';
import { resolve } from 'path';
import { env } from 'process';

import {
  babelJitEnv,
  modulePath,
  requireBabelJitArgs,
  requireTypescriptJitArgs,
  typescriptJitEnv,
} from '../utils';

const mochaCli = resolve(modulePath('mocha'), 'bin', 'mocha');

export const runMocha = (cwd: string) => {
  const { status, stdout } = run(
    'node',
    [
      mochaCli,
      ...requireBabelJitArgs,
      ...requireTypescriptJitArgs,
      '--reporter',
      'json',
      'mocha.js',
    ],
    {
      cwd,
      env: {
        ...env,
        ...babelJitEnv,
        ...typescriptJitEnv,
      },
    },
  );

  return { status, result: JSON.parse(stdout.toString()) };
};
