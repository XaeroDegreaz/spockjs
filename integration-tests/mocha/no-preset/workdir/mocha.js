specify('successful test', () => {
  expect: 1 === 1;
});

specify('failing test', () => {
  let x;
  const double = n => n * 2;

  when: x = 3;
  then: x === double(2);
});
