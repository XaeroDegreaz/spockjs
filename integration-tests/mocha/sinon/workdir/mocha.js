import assert from 'assert';
import * as sinon from 'sinon';

specify('stubbing', () => {
  const s = sinon.stub();
  stub: s() >> 42;
  assert(s() === 42);
});

specify('mocking', () => {
  const obj = { method: () => {} };

  const mock = sinon.mock(obj);
  mock: 2 * mock.method();

  obj.method();
  obj.method();

  verify: mock;
});

specify('failing verification', () => {
  const obj = { method: () => {} };

  const mock = sinon.mock(obj);
  mock: 2 * mock.method();

  obj.method();

  verify: mock;
});
