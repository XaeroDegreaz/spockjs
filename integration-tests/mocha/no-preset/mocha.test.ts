import { resolve } from 'path';

import { runMocha } from '../mocha';

// mark implicit dependencies for jest
() => require('./workdir/mocha.js') && require('./workdir/package.json');

const cwd = resolve(__dirname, 'workdir');

test('produces correct output', () => {
  const {
    status,
    result: {
      stats: { passes, failures },
      failures: [
        {
          err: { message },
        },
      ],
    },
  } = runMocha(cwd);

  expect: status === 1;

  expect: {
    passes === 1;
    failures === 1;
  }
  expect(message).toMatchSnapshot();
});
