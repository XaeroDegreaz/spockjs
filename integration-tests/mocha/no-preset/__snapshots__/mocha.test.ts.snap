// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`produces correct output 1`] = `
"(x === double(2)) is not truthy   # mocha.js:10
  
  _assert(x === double(2), \\"(x === double(2)) is not truthy\\")
          | |   |                                            
          | |   4                                            
          3 false                                            
  
  [number] double(2)
  => 4
  [number] x
  => 3
  "
`;
